pkg/controllers/provisioningv2/rke2/provisioningcluster/template.go

```diff
@@ -45,6 +46,10 @@ func getInfraRef(rkeCluster *rkev1.RKECluster) *corev1.ObjectReference {
 // objects generates the corresponding rkecontrolplanes.rke.cattle.io, clusters.cluster.x-k8s.io, and
 // machinedeployments.cluster.x-k8s.io objects based on the passed in clusters.provisioning.cattle.io object
 func objects(cluster *rancherv1.Cluster, dynamic *dynamic.Controller, dynamicSchema mgmtcontroller.DynamicSchemaCache, secrets v1.SecretCache) (result []runtime.Object, _ error) {
+	if !cluster.DeletionTimestamp.IsZero() {
+		return nil, nil
```

Comment:
Not really a problem but I feel like this can be short-circuited further upwards. Probably good to have this check in here anyway